### DISCUSSION TITLE
Use PythonCall in a package to call from a pre-existing conda environment

### DISCUSSION THREAD

**Discussion by gabriorizzuti** [cjdoris/PythonCall.jl#560] [Q&A]
Hello everyone!

I’m trying to develop a glue package that exposes some Python functionalities to Julia, and playing around with PythonCall.jl.

I do have a pre-existing conda environment and I’d like to point PythonCall to this environment. So far I was unable to make progress, and I was hoping for some tips.

This is what I tried, after setting an environment variable PYTHON_IN_SOME_CONDA_ENV that points to the afore-mentioned pre-existing conda environment python exe:
```
module PyPkg4Julia

ENV["JULIA_CONDAPKG_BACKEND"] = "Null"
ENV["JULIA_PYTHONCALL_EXE"] = ENV["PYTHON_IN_SOME_CONDA_ENV"]
using PythonCall

const pypkg = Ref{Py}()
function __init__()
     const pypkg[] = pyimport("some_pkg_in_cond_env")
end

export fun_wrapper
fun_wrapper(x) = pypkg[].fun(x)

end
```

I get the following error `ERROR: Python: ModuleNotFoundError: No module named 'some_pkg_in_cond_env'`.
I'm pretty sure there are problems with setting the JULIA_PYTHONCALL_EXE, but not so sure how to fix it.

Any help is extremely appreciated!

**Comment by cjdoris** [comment]
The lines
```
ENV["JULIA_CONDAPKG_BACKEND"] = "Null"
ENV["JULIA_PYTHONCALL_EXE"] = ENV["PYTHON_IN_SOME_CONDA_ENV"]
```
are only run once, when `PyPkg4Julia` is installed. They are not run again when it is imported, so you end up using CondaPkg's default environment.

The settings `JULIA_PYTHONCALL_EXE` etc. are user-level config. They are not intended to be set from within packages. If a user wants to use a pre-existing environment they should set those env cars themselves before importing your package.
